ext/em.cpp

```diff
@@ -877,7 +883,7 @@ _SelectDataSelect
 static VALUE _SelectDataSelect (void *v)
 {
 	SelectData_t *sd = (SelectData_t*)v;
-	sd->nSockets = rb_fd_select (sd->maxsocket+1, &(sd->fdreads), &(sd->fdwrites), &(sd->fderrors), &(sd->tv));
+	sd->nSockets = select (sd->maxsocket+1, &(sd->fdreads), &(sd->fdwrites), &(sd->fderrors), &(sd->tv));
```

Comment:
This is failing to compile since we're no longer using raw `fd_set`s. Switching to `select` here also reintroduces the original issue where the select reactor breaks on large fds.

Unfortunately it doesn't look like rb_fd_select is safe to call from outside the gvl, because it can invoke ruby_xmalloc which can kick off a ruby GC.

The correct solution here is probably to prefer `rb_fd_select` over the raw `rb_thread_blocking_region` and `rb_thread_call_without_gvl` apis when available, to let the VM do all the blocking GVL stuff for us.